point-of-sale-app/inventory/src/main/java/com/google/abmedge/inventory/dao/InMemoryStoreConnector.java

```diff
@@ -0,0 +1,130 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.abmedge.inventory.dao;
+
+import com.google.abmedge.inventory.dto.Item;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * An implementation of the {@link InventoryStoreConnector} that uses an in memory data structure to
+ * hold all the inventory information. This implementation uses a simple HashMap to store all the
+ * items
+ */
+public class InMemoryStoreConnector implements InventoryStoreConnector {
+
+  private static final Logger LOGGER = LogManager.getLogger(InMemoryStoreConnector.class);
+  private static final Map<UUID, Item> idToItemsMap = new HashMap<>();
+
+  @Override
+  public List<Item> getAll() {
+    return idToItemsMap.values()
+        .stream()
+        .map(Item::from)
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public List<Item> getAllByType(String type) {
+    return idToItemsMap.values()
+        .stream()
+        .filter(i -> i.getType().equals(type))
+        .map(Item::from)
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public Optional<Item> getById(UUID id) {
+    return idToItemsMap.values()
+        .stream()
+        .filter(i -> i.getId().equals(id))
+        .map(Item::from)
+        .findAny();
+  }
+
+  @Override
+  public boolean insert(Item item) {
+    UUID itemId = item.getId();
+    String itemType = item.getType();
+    if (itemId == null || StringUtils.isEmpty(itemType)) {
+      LOGGER.error("Cannot insert item. Item must have both 'id' and 'type'. "
+          + "Passed in item is missing one or both of these attributes");
+      return false;
```

Comment:
Throwing Exceptions is a better way than returning a boolean flag.

Reply:
Replaced boolean return with exception in 0f9e7029c50f6f46bed275962f5687a345e11a4d